### PR TITLE
feat: #9 Check from hospital if drugstore has enough of certain drug

### PR DESCRIPTION
**Pull request by lazar-stojcevic**
There is need of implementation of more test for this feature but this feature is blocking one other, so not all test are written.

### CODE REVIEW

File: src/DrugstoreApp/Backend/DrugstoreAPI/Drugstore/Repository/Interfaces/IFeedbackRepository.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Drugstore.Repository.Interfaces
+{
+    public interface IFeedbackRepository
+    {
+        public bool Delete(int id);
+        public List<Feedback> GetAll();
+        public bool Save(Feedback newFeedback);
```
Comment (Aleksandar9999) [inline src/DrugstoreApp/Backend/DrugstoreAPI/Drugstore/Repository/Interfaces/IFeedbackRepository.cs:12]: Save method should return Feedback object created in db.

File: src/DrugstoreApp/Backend/DrugstoreAPI/Drugstore/Repository/Interfaces/IHospitalRepository.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace Drugstore.Repository.Interfaces
+{
+    public interface IHospitalRepository
+    {
+        public List<Hospital> GetAll();
+        public bool GetHospitalById(int id);
+        public int GetIdByName(string name);
+        public string GetKeyByName(string name);
+        public bool Save(Hospital newHospital);
```
Comment (Aleksandar9999) [inline src/DrugstoreApp/Backend/DrugstoreAPI/Drugstore/Repository/Interfaces/IHospitalRepository.cs:14]: Also Save should return object created in db.

File: src/DrugstoreApp/Backend/DrugstoreAPI/Drugstore/Service/MedicineService.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            MedicineRepository = medicineRepository;
+        }
+
+        public MedicineService()
+        {
+            MedicineRepository = new MedicineSqlRepository();
+        }
+
+        public bool CheckForAmountOfDrug(string nameOfDrug, int amountOfDrug)
```
Comment (Aleksandar9999) [inline src/DrugstoreApp/Backend/DrugstoreAPI/Drugstore/Service/MedicineService.cs:24]: Please refactor this method

File: src/Integration API/Controllers/DrugPurchaseController.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        private readonly MyDbContext dbContext;
+
+        public DrugPurchaseController(MyDbContext db)
+        {
+            this.dbContext = db;
+        }
+
+
+        [HttpPut]
+        public IActionResult Put(DrugAmountDemandDto demand)
```
Comment (Aleksandar9999) [inline src/Integration API/Controllers/DrugPurchaseController.cs:27]: Think about refactoring this method

File: src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Controllers/DrugDemandController.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                    {
+                       return Ok(medicineService.CheckForAmountOfDrug(demand.Name, demand.Amount));
+                    }
+                }
+            }
+            return Unauthorized();
+        }
+
+
+        public bool checkApiKey(string apiKey, MyDbContext dbContext)
```
Comment (Aleksandar9999) [inline src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Controllers/DrugDemandController.cs:48]: Please refactor this methode.

**Review by Aleksandar9999 — CHANGES_REQUESTED** [review]
Also, think about using some .net core library for authorization.

File: src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Controllers/DrugDemandController.cs (hunk 5 above)
Comment (lenche23) [reply src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Controllers/DrugDemandController.cs:48]: method*

File: src/DrugstoreApp/Backend/DrugstoreAPI/Drugstore/Repository/Interfaces/IFeedbackRepository.cs (hunk 1 above)
Comment (lazar-stojcevic) [reply src/DrugstoreApp/Backend/DrugstoreAPI/Drugstore/Repository/Interfaces/IFeedbackRepository.cs:12]: I will use Command-Query separation pattern, so it's gonna be void!

File: src/DrugstoreApp/Backend/DrugstoreAPI/Drugstore/Repository/Interfaces/IHospitalRepository.cs (hunk 2 above)
Comment (lazar-stojcevic) [reply src/DrugstoreApp/Backend/DrugstoreAPI/Drugstore/Repository/Interfaces/IHospitalRepository.cs:14]: I will use Command-Query separation pattern, so it's gonna be void!

File: src/DrugstoreApp/Backend/DrugstoreAPI/Drugstore/Service/MedicineService.cs (hunk 3 above)
Comment (lazar-stojcevic) [reply src/DrugstoreApp/Backend/DrugstoreAPI/Drugstore/Service/MedicineService.cs:24]: OK

File: src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Controllers/DrugDemandController.cs (hunk 5 above)
Comment (lazar-stojcevic) [reply src/DrugstoreApp/Backend/DrugstoreAPI/DrugstoreAPI/Controllers/DrugDemandController.cs:48]: I don't know why is this method was public or why was I sending dbContext but now is ok I guess.

File: src/Integration API/Controllers/DrugPurchaseController.cs (hunk 4 above)
Comment (lazar-stojcevic) [reply src/Integration API/Controllers/DrugPurchaseController.cs:27]: ok

**Review by Aleksandar9999 — APPROVED** [review]

**Review by lenche23 — APPROVED** [review]

**Pull request merged**